### PR TITLE
PLAT-1799: Table: Make private props readonly

### PR DESCRIPTION
**Pull request by chavlji**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1799)

3rd part of ticket [PLAT-1667: Table: indeterminate state inconsistent](https://reciprocitylabs.atlassian.net/browse/PLAT-1667)

**!WARNING!**
Based on https://github.com/reciprocity/zen-ui/pull/309! It should be merged first!

#### Description
As per JIRA ticket

### CODE REVIEW

File: src/components/zen-table-cell/zen-table-cell.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
+import { html } from 'lit-html';
+import { getArgTypes, getDefaultArgs, getComponentData, spreadArgs } from '../../../.storybook/helpers/argTypes';
+
+import data from '../../../stencilDocs.json';
+const compData = data.components.find(n => n.tag === 'zen-table-cell');
+const argTypes = getArgTypes(compData);
+// disable props that are controlled by table and would have no impact:
+// argTypes.$visible.control = { type: 'none' };
+// argTypes.$expandable.control = { type: 'none' };
```
Comment (homer0) [inline src/components/zen-table-cell/zen-table-cell.stories.mdx:10]: Shouldn't this be uncommented?

File: src/components/zen-table-row/test/zen-table-row.spec.tsx [hunk 2]
```diff
@@ -55,20 +55,20 @@ describe('zen-table-row', () => {
     await page.waitForChanges();
     await cleanupTableStructure(table);
 
-    expect(mainRow.visible).toBeTruthy();
+    expect(mainRow.$visible).toBeTruthy();
```
Comment (homer0) [inline src/components/zen-table-row/test/zen-table-row.spec.tsx:58]: Not a blocker, just something to have in mind: when validating booleans, you should use `.toBe(true)` as it's more specific; `[]` is truthy :P.

**Review by homer0 — APPROVED** [review]
I didn't see any red flags on the code and the functionality is working as expected on Storybook 🤘.

A couple of things:

- The comment in the cell story is a question, if `false`, just delete them :P.
- The thing about `toBe(true)` is just to have it in mind, no need to change it right now.
- It's ok since everyone is aware of the state of the table (being that we are still working on it), but...
 - The `refactor` commit should've had a breaking change comment, as you removed props.
 - The `chore` for the style should've been a `fix`; it could be a `chore` if the code was new to the PR, but you were actually changing something that already existed.

Awesome work 🤘

**Review by mihaelpalfi — APPROVED** [review]
Good. Really tried to find something wrong on all browsers, no luck. Just maybe a NIT : 
Nothing to do with this ticket. The known issue that in tree the first row selected has no top blue border. 

image

File: src/components/zen-table-cell/zen-table-cell.stories.mdx (hunk 1 above)
Comment (chavlji) [reply src/components/zen-table-cell/zen-table-cell.stories.mdx:10]: Removed it altogether as all props have some impact on the story

**Comment by chavlji** [comment]
@homer0 tests fixed. Things regarding commit messages noted.
@mihaelpalfi fixed.

**Pull request merged**